test/blackbox-tests/test-cases/pkg/post-deps-solving.t

```diff
@@ -0,0 +1,86 @@
+Solving for post dependencies:
+
+  $ . ./helpers.sh
+  $ mkrepo
+
+  $ mkpkg bar
+
+  $ mkpkg foo <<EOF
+  > depends: [ "bar" {post} ]
+  > EOF
+
+We don't need bar, but we still include it:
+
+  $ solve foo
```

Comment:
We shouldn't lock packages whose sources we aren't actually using for the build. We can keep this undesirable property for now until we figure out how post deps actually affect the build.